.github/workflows/build-containers.yml

```diff
@@ -1,60 +1,152 @@
 name: Build Containers
 
 on:
+  pull_request:
+    paths:
+      - .github/workflows/build-containers.yml
+      - devops/docker/**/Dockerfile
   push:
-    branches: [main, future/**]
-    paths: [devops/docker/**/Dockerfile]
+    branches:
+      - main
+    paths:
+      - .github/workflows/build-containers.yml
+      - devops/docker/**/Dockerfile
+
+env:
+  BUILD_TYPE: Release
+  MOUNT: /azure-osconfig
 
 jobs:
-  list-containers:
-    name: List modified containers
+  modified-containers:
+    name: Modified containers
     runs-on: ubuntu-latest
     outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix }}
+      containers: ${{ steps.matrix.outputs.containers }}
     steps:
       - uses: actions/checkout@v2
 
-      - name: Get changed files
-        id: changed-files
+      - name: Get changed containers
+        id: changed-containers
         uses: tj-actions/changed-files@v18.7
         with:
           files: ./devops/docker/**/Dockerfile
 
-      - id: set-matrix
-        run: echo "matrix=$(echo -n ${{ steps.changed-files.outputs.all_changed_files }} | jq -R -s -c 'split(" ")')" >> $GITHUB_OUTPUT
+      - name: Create build matrix
+        id: matrix
+        run: |
+          # If this workflow is present in other_changed_files, then rebuild all containers
+          if [[ "${{ steps.changed-containers.outputs.other_changed_files }}" == *".github/workflows/build-containers.yml"* ]]; then
+            echo "Workflow file changed, adding all containers to build matrix..."
+            containers=$(ls -d devops/docker/**/Dockerfile | jq -R . | jq -s .)
+          else
+            containers=$(echo -n ${{ steps.changed-containers.outputs.all_changed_files }} | jq -R -s -c 'split(" ")')
+          fi
+
+          echo Containers to build: $containers
+          echo containers=$containers >> $GITHUB_OUTPUT
 
-  docker:
-    name: Build and push
-    needs: list-containers
-    runs-on: [self-hosted, 1ES.Pool=azure-osconfig-linux-pool]
+  docker-build:
+    name: Docker
+    needs: modified-containers
+    runs-on: ubuntu-latest
```

Comment:
Were using github-hosted runners here, we need to have ES runners to ensure containers are built on ES systems.

Reply:
Agreed, I have temporarily skipped (`if: false`) the steps that would push to our ACR since we are pushing the work related to 1ES runners. In the mean time these changes will remove the secrets from CI workflows. When the work to publish from GitHub is done (using 1ES runners) this and other jobs will be updated with the correct pools/1ES runners.